src/error-boundary/internal.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { Component, createContext, useContext, useState } from 'react';
+
+import { useMergeRefs } from '@cloudscape-design/component-toolkit/internal';
+
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import { SomeRequired } from '../internal/types';
+import { ErrorBoundaryFallback } from './fallback';
+import { ErrorBoundaryProps } from './interfaces';
+
+import styles from './styles.css.js';
+
+// Helper methods attached to the component's root node for e2e testing.
+declare global {
+  interface HTMLElement {
+    __awsui__?: { forceError?(): void; clearForcedError?(): void };
+  }
+}
+
+const RootSuppressed = Symbol();
+
+const ErrorBoundariesContext = createContext<
+  Pick<ErrorBoundaryProps, 'renderFallback' | 'i18nStrings' | 'onError' | 'errorBoundaryId'> & {
+    suppressed: boolean | typeof RootSuppressed;
+  }
+>({
+  onError: () => {},
+  suppressed: RootSuppressed,
+});
+
+interface InternalErrorBoundaryProps
+  extends SomeRequired<ErrorBoundaryProps, 'suppressNested'>,
+    InternalBaseComponentProps {}
+
+export function InternalErrorBoundary({
+  children,
+  suppressNested,
+  __internalRootRef,
+  ...props
+}: InternalErrorBoundaryProps) {
+  const context = useContext(ErrorBoundariesContext);
+  const thisSuppressed = context.suppressed === true;
+  const nextSuppressed = suppressNested || thisSuppressed;
+
+  const [forcedError, setForcedError] = useState(false);
+  const setElement = (node: null | HTMLDivElement) => {
+    if (node) {
+      if (!node.__awsui__) {
+        node.__awsui__ = {};
+      }
+      node.__awsui__.forceError = () => setForcedError(true);
+      node.__awsui__.clearForcedError = () => setForcedError(false);
+    }
+  };
+  const mergedRef = useMergeRefs(setElement, __internalRootRef);
+
+  return (
+    <div ref={mergedRef} className={styles['error-boundary']} data-awsui-boundary-id={props.errorBoundaryId}>
+      {!thisSuppressed ? (
+        <ErrorBoundaryImpl {...props} forcedError={forcedError}>
+          <ErrorBoundariesContext.Provider value={{ ...props, suppressed: nextSuppressed }}>
+            {children}
+          </ErrorBoundariesContext.Provider>
+        </ErrorBoundaryImpl>
+      ) : (
+        <>{children}</>
```

Comment:
Nitpick: could this

```
(
  <>{children}</>
)
```

be reduced to
```
children
```
?

Reply:
Thanks, I will check TS is happy with that and update if possible.